pkg/kafka/auth/plaintext.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"github.com/Shopify/sarama"
+)
+
+// PlainTextConfig describes the configuration properties needed for SASL/PLAIN with kafka
+type PlainTextConfig struct {
+	UserName string
+	Password string
+}
+
+func setPlainTextConfiguration(config *PlainTextConfig, saramaConfig *sarama.Config) {
+	saramaConfig.Net.SASL.Enable = true
+	saramaConfig.Net.SASL.User = config.UserName
+	saramaConfig.Net.SASL.Password = config.Password
```

Comment:
this doesn't track. Kafka docs https://docs.confluent.io/current/kafka/authentication_sasl/authentication_sasl_plain.html mention "plaintext" as an alternative to SASL "plain", but here we seem to make them the same.

Reply:
Please correct me if I understood it, You are trying to say rather than using "PlainText" we shall use "Plain".

Reply:
no, I am saying Kafka docs make clear distinction between those two methods, but in this code we seem to treat them as the same thing: calling it plaintext but using SASL structure to pass it. So which one is it?

Reply:
saramaconfig only has SASL struct. below is the code snippet.

```
// SASL based authentication with broker. While there are multiple SASL authentication methods
		// the current implementation is limited to plaintext (SASL/PLAIN) authentication
		SASL struct {
			// Whether or not to use SASL authentication when connecting to the broker
			// (defaults to false).
			Enable bool
			// SASLMechanism is the name of the enabled SASL mechanism.
			// Possible values: OAUTHBEARER, PLAIN (defaults to PLAIN).
			Mechanism SASLMechanism
			// Whether or not to send the Kafka SASL handshake first if enabled
			// (defaults to true). You should only set this to false if you're using
			// a non-Kafka SASL proxy.
			Handshake bool
			//username and password for SASL/PLAIN  or SASL/SCRAM authentication
			User     string
			Password string
```

Reply:
well, then this is "plain", not "plaintext", because credentials are still sent over encrypted channel. From the docs:

> `PLAIN` versus `PLAINTEXT` Do not confuse the _SASL mechanism PLAIN_ with _no SSL encryption being called PLAINTEXT_.

Reply:
From Confluent slack channel:
`None of the SASL mechanisms encrypt wire data not even GSSAPI. You have to enable TLS for that.`

Reply:
https://docs.confluent.io/current/kafka/authentication_ssl.html
> By default, Apache Kafka® communicates in PLAINTEXT, which means that all data is sent in the clear. To encrypt communication, it is recommended to configure all the Confluent Platform components in your deployment to use SSL encryption.